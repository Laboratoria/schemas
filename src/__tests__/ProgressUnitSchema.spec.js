const mongoose = require('mongoose/browser');
const {
  ProgressUnitSchema, TopicUnitSchema,
} = require('../..')(mongoose);

describe('ProgressUnitSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, ProgressUnitSchema);
    return doc.validate()
      .catch(err => expect(err.errors).toMatchSnapshot());
  });

  it('should pass at least with unit field', () => {
    const topicUnit = new mongoose.Document({}, TopicUnitSchema);
    const doc = new mongoose.Document({
      unit: topicUnit._id,
    }, ProgressUnitSchema);

    return doc.validate();
  });

  it('should pass with all fields', () => {
    const topicUnit = new mongoose.Document({}, TopicUnitSchema);
    const doc = new mongoose.Document({
      unit: topicUnit._id,
      completedDuration: 20,
      completedParts: 5,
      percent: 100,
      completedAt: Date.now(),
    }, ProgressUnitSchema);

    return doc.validate();
  });
});
