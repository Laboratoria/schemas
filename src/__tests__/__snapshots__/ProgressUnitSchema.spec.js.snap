// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressUnitSchema should fail validation when fields missing 1`] = `
Object {
  "unit": [ValidatorError: Path \`unit\` is required.],
}
`;
